### DISCUSSION TITLE
How to get newest error state in onChange?

### DISCUSSION THREAD

**Discussion by mterczynski** [react-hook-form/react-hook-form#8697] [Q&A] [answered]
Hi, I would like to listen to input changes and call a function that has the access to newest input value and the newest error state.

The problem that I'm facing is that the `errors` and `state` are outdated. For example, if you type `a`, `onChange` logs `state.invalid` as `false` because it shows `state.invalid` for the previous input value.
I've also tried to use `useEffect` instead of `onChange` but I get the same results. I would appreciate any help/ideas. 

Code sandbox:
https://codesandbox.io/s/cranky-goldberg-zheos1?file=/src/App.js

**Comment by Aditya000786** [comment, marked as answer]
I think u can react to formState change by putting it in an optional array of useEffect hook.

**Reply by bluebill1049** [reply]
thatnks @Aditya000786 yes, formState update will always update via state.
